QModManager/Patching/StoreDetector.cs

```diff
@@ -0,0 +1,97 @@
+﻿namespace QModManager.Patching
+{
+    using System;
+    using System.IO;
+    using Checks;
+
+    internal class StoreDetector
+    {
+        internal static string GetUsedGameStore()
+        {
+            string directory = null;
+
+            try
+            {
+                directory ??= Environment.CurrentDirectory;
+            }
+            catch
+            {
+                return "Error on getting Store";
+            }
+
+            if (NonValidStore(directory))
+            {
+                return "free Store";
+            }
+            else if (IsSteam(directory))
+            {
+                return "Steam";
+            }
+            else if (IsEpic(directory))
+            {
+                return "Eic Games";
+            }
+            else if (IsMSStore(directory))
+            {
+                return "MSStore";
+            }
+            else
+            {
+                return "was not able to identify Store";
+            }
+        }
+
+        internal static bool IsSteam(string directory)
+        {
+            string checkfile = Path.Combine(directory, "steam_api64.dll");
+            if (File.Exists(checkfile))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        internal static bool IsEpic(string directory)
+        {
+            string checkfolder = Path.Combine(directory, ".eggstore");
+            if (Directory.Exists(checkfolder))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        internal static bool IsMSStore(string directory)
+        {
+            string checkfile = Path.Combine(directory, "MicrosoftGame.config");
+            if (File.Exists(checkfile))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        internal static bool NonValidStore(string folder)
```

Comment:
Would suggest renaming this to `InvalidStore` rather than `NonValidStore`. If you accept this suggestion, make sure to update the calls!

```suggestion
        internal static bool InvalidStore(string folder)
```